test/e2e/e2e_test.go

```diff
@@ -147,8 +147,6 @@ func pingTestJobSet(ns *corev1.Namespace) *testing.JobSetWrapper {
 	return testing.MakeJobSet(jsName, ns.Name).
 		EnableDNSHostnames(true).
 		NetworkSubdomain(jsName).
```

Comment:
lets remove this too to test the case where the subdomain is not explicitly set.